File: src/Application Specific/Classes/Entity Collections/Authority/SiteAuthority.cs
using System.Collections;

namespace FSM.Entity
{
    namespace Authority
    {
        public class SiteAuthority
        {
            private DataTypeValidator _dataTypeValidator;

            public SiteAuthority()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            private int _siteAuthorityID = 0;

            public int SiteAuthorityID
            {
                get
                {
                    return _siteAuthorityID;
                }
            }

            public object SetSiteAuthorityID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_siteAuthorityID", value);
                }
            }

            private int _siteId = 0;

            public int SiteID
            {
                get
                {
                    return _siteId;
                }
            }

            public object SetSiteID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_siteId", value);
                }
            }

            private int _authorityID = 0;

            public int AuthorityID
            {
                get
                {
                    return _authorityID;
                }
            }

            public object SetAuthorityID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_authorityID", value);
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Authority/SiteAuthority.vb
Imports System.Data.SqlClient
Imports System.Reflection

Namespace Entity

    Namespace Authority

        Public Class SiteAuthority

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "Property Properties"

            Private _siteAuthorityID As Integer = 0
            Public ReadOnly Property SiteAuthorityID() As Integer
                Get
                    Return _siteAuthorityID
                End Get
            End Property
            Public WriteOnly Property SetSiteAuthorityID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_siteAuthorityID", Value)
                End Set
            End Property

            Private _siteId As Integer = 0
            Public ReadOnly Property SiteID() As Integer
                Get
                    Return _siteId
                End Get
            End Property
            Public WriteOnly Property SetSiteID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_siteId", Value)
                End Set
            End Property

            Private _authorityID As Integer = 0
            Public ReadOnly Property AuthorityID() As Integer
                Get
                    Return _authorityID
                End Get
            End Property
            Public WriteOnly Property SetAuthorityID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_authorityID", Value)
                End Set
            End Property

#End Region

        End Class

    End Namespace

End Namespace

